src/transport/manager/address.rs

```diff
@@ -18,12 +18,29 @@
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 // DEALINGS IN THE SOFTWARE.
 
-use crate::{types::ConnectionId, PeerId};
+use crate::{error::DialError, types::ConnectionId, PeerId};
 
 use multiaddr::{Multiaddr, Protocol};
 use multihash::Multihash;
 
-use std::collections::{BinaryHeap, HashSet};
+use std::collections::{hash_map::Entry, HashMap};
+
+/// Maximum number of addresses tracked for a peer.
+const MAX_ADDRESSES: usize = 64;
+
+/// Scores for address records.
+pub mod scores {
+    /// Score indicating that the connection was successfully established.
+    pub const CONNECTION_ESTABLISHED: i32 = 100i32;
+
+    /// Score for failing to connect due to an invalid or unreachable address.
+    pub const CONNECTION_FAILURE: i32 = -100i32;
+
+    /// Score for providing an invalid address.
+    ///
+    /// This address can never be reached.
+    pub const ADDRESS_FAILURE: i32 = 5 * CONNECTION_FAILURE;
```

Comment:
?
```suggestion
    pub const ADDRESS_FAILURE: i32 = i32::MIN;
```

Reply:
Yep that makes sense, since the address will not be reached